LANDSCAPE/COMPARISONS/ARGOCD-FLUX/index.md

```diff
@@ -0,0 +1,275 @@
+
+
+## Flux CD
+
+FluxCD [installation/bootstrapping](https://fluxcd.io/docs/cmd/flux_bootstrap/) mechanism can be done together. The prerequisites for bootstrapping can be found in the [upstream documentation](https://fluxcd.io/docs/get-started/). During bootstrapping, Flux will create a repository on GitHub or GitLab (A generic “git” mode can also be used) and then create all the manifests for installation and connection to the git repo in that same git repo. The Flux installer will then “hand off” the responsibility of managing/syncing the cluster to the_ flux-system_ that is installed on the cluster, pointing to the repo that was just created.
+
+Flux can also be installed by [just using manifests](https://fluxcd.io/docs/installation/#dev-install), for those that don’t want to use bootstrapping.
+
+
+## Argo CD
+
+Argo CD has no native mechanism for bootstrapping. The [install](https://argo-cd.readthedocs.io/en/stable/getting_started/#1-install-argo-cd) is just that: the install of the Argo CD components. It leaves the rest of the configuration up to the end user. The install is straightforward. It is done via a YAML file using the `kubectl apply` command on said YAML files. Once Argo CD is installed, it’s ready to be configured and used.
+
+There is a project called “[Argo CD Autopilot](https://github.com/argoproj-labs/argocd-autopilot)” upstream, which is used to bootstrap Argo CD onto a cluster. It also takes an opinionated approach to how your git directory structure should look like. This is an open source project created by [Codefresh](codefresh.io/).
+
+
+# Reconciliation Configurability
+
+Both projects offer a way to configure how reconciliation works. In this section we’ll focus specifically on the differences between configuring the reconciliation time.
+
+
+## FluxCD
+
+Flux v2 is built on the [GitOps Toolkit](https://fluxcd.io/docs/components/), which has different components. The Flux reconciliation loop can be configured at each [component](https://fluxcd.io/docs/components/). For example you can have your git repository checked every 5 minutes, but then do the sync every 10 minutes. This gives you the ability to stagger how the reconciliation happens.
+
+
+## Argo CD
+
+Argo CD reconciliation is a global setting. This is one setting that can be modified via the [Argo CD configuration ConfigMap](https://github.com/argoproj/argo-cd/blob/master/docs/operator-manual/argocd-cm.yaml#L240). By default, this is set to 3 minutes and can be changed by the end user. Although you have a choice of disabling reconciliation globally or per-application, you can only set the reconciliation time globally. This means you cannot set the reconciliation time differently for each [Application](https://argo-cd.readthedocs.io/en/stable/core_concepts/).
```

Comment:
```suggestion
Argo CD reconciliation is a global setting. This is one setting that can be modified via the [Argo CD configuration ConfigMap](https://github.com/argoproj/argo-cd/blob/891d9b2f380e1879474ab57c1adf21f5d1678759/docs/operator-manual/argocd-cm.yaml#L246). By default, this is set to 3 minutes and can be changed by the end user. Although you have a choice of disabling reconciliation globally or per-application, you can only set the reconciliation time globally. This means you cannot set the reconciliation time differently for each [Application](https://argo-cd.readthedocs.io/en/stable/core_concepts/).
```

Reply:
@christianh814 et al.

❔ **Do Argo CD and Flux differ at all in the nuances of:**
• Drift Detection and Remediation
• Continuous Operations (CO)
?

❔ To answer that, would you run tests with their latest versions to compare and contrast?

**For example:** If a person runs `kubectl` to manually overwrite Kubernetes cluster's actual state, do Argo CD and Flux _(not only)_ monitor the runtime environment to notice the drift between actual state and desired state, and also converge them to remediate automatically?

My general understanding is that Argo CD and Flux can be used for drift detection and remediation, and Continuous Operations (CO).

**References:**

**[Cloud-native Patterns - Ops Edition - 00:00](https://www.youtube.com/watch?v=fZEFWh4xBEw)**

<img width="1440" alt="Cloud-native Patterns - Ops Edition - 00:00" src="https://user-images.githubusercontent.com/1329685/144722421-5974824e-b70e-485c-abb8-d30fdbfcae97.png">

**[Cloud-native Patterns - Ops Edition - 19:53 - Continuous Operations - Monitor Runtime Env](https://www.youtube.com/watch?v=fZEFWh4xBEw&t=1193s)**

<img width="1440" alt="Cloud-native Patterns - Ops Edition - 19:53 - Continuous Operations - Monitor Runtime Env" src="https://user-images.githubusercontent.com/1329685/144722436-cdaf6d99-bafe-448b-aac7-a94297c8e88d.png">

**[Cloud-native Patterns - Ops Edition - 37:45 - Drift detection and remediation](https://www.youtube.com/watch?v=fZEFWh4xBEw&t=2265s)**

<img width="1440" alt="Cloud-native Patterns - Ops Edition - 37:45 - Drift detection and remediation" src="https://user-images.githubusercontent.com/1329685/144722446-c5d13cb8-79a6-4681-91a8-cc910e181900.png">

**[Cloud-native Patterns - Ops Edition - 42:03 - Continuous Operations - Monitor Runtime Env](https://www.youtube.com/watch?v=fZEFWh4xBEw&t=2523s)**

<img width="1440" alt="Cloud-native Patterns - Ops Edition - 42:03 - Continuous Operations - Monitor Runtime Env" src="https://user-images.githubusercontent.com/1329685/144722458-c60759fa-bb88-4808-b854-5d8bbca8e15a.png">

**Thank you.**

Reply:
[Argo CD `timeout.reconciliation`](https://github.com/argoproj/argo-cd/blob/master/docs/operator-manual/argocd-cm.yaml#L243:L246)
[#L240](https://github.com/argoproj/argo-cd/blob/master/docs/operator-manual/argocd-cm.yaml#L240) → [#L243:L246](https://github.com/argoproj/argo-cd/blob/master/docs/operator-manual/argocd-cm.yaml#L243:L246)
```suggestion
Argo CD reconciliation is a global setting. This is one setting that can be modified via the [Argo CD configuration ConfigMap](https://github.com/argoproj/argo-cd/blob/master/docs/operator-manual/argocd-cm.yaml#L243:L246). By default, this is set to 3 minutes and can be changed by the end user. Although you have a choice of disabling reconciliation globally or per-application, you can only set the reconciliation time globally. This means you cannot set the reconciliation time differently for each [Application](https://argo-cd.readthedocs.io/en/stable/core_concepts/).
```

Reply:
Argo definitely hand slaps you for using k apply -f manually and sets it back to the last declared state in the Git repo that's sync'ed